src/components/NoteCard.jsx

```diff
@@ -32,6 +47,50 @@ const NoteCard = ({ notes }) => {
     }
   };
 
+  const addToArchive = async (notes) => {
+    try {
+      const { data, status } = await addToArchiveOperation(notes, encodedToken);
+      if (status === 200 || status === 201) {
+        noteDispatch({
+          type: "ADD_TO_ARCHIVE",
+          payload: { notes: data.notes, archives: data.archives },
+        });
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  const unArchiveNote = async (notes) => {
+    try {
+      const { data, status } = await unarchiveNote(notes, encodedToken);
+      if (status === 200 || status === 201) {
+        noteDispatch({
+          type: "ADD_TO_ARCHIVE",
+          payload: { notes: data.notes, archives: data.archives },
+        });
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  const deleteFromArchiveFunc = async (notes) => {
```

Comment:
all of these functions can go to a separate utills file

Reply:
Yes will refactor code